src/main/ruby/truffleruby/core/string.rb

```diff
@@ -529,11 +529,11 @@ def inspect
   private def inspect_char(enc, result_encoding, ascii, unicode, index, char, result)
     consumed = char.bytesize
 
-    if (ascii or unicode) and consumed == 1
+    if (ascii || unicode) || consumed == 1
```

Comment:
I figured `||` is preferred over `or` and that's the style of the rest of `string.rb`

Reply:
Yes should be `||` in these cases.